test/e2e/framework/auth/helpers.go

```diff
@@ -45,11 +45,12 @@ type bindingsGetter interface {
 
 // WaitForAuthzUpdate checks if the give user can perform named verb and action
 // on a resource or subresource.
-func WaitForAuthzUpdate(ctx context.Context, c v1authorization.SubjectAccessReviewsGetter, user string, ra *authorizationv1.ResourceAttributes, allowed bool) error {
+func WaitForAuthzUpdate(ctx context.Context, c v1authorization.SubjectAccessReviewsGetter, user string, groups []string, ra *authorizationv1.ResourceAttributes, allowed bool) error {
```

Comment:
just a thought: maybe expecting everyone to add the right groups is too much?

Could we in this function detect that the user had prefix system:serviceaccount: and add those groups automatically.

I don't feel strongly about this so feel free to ignore this suggestion.

Reply:
IMO we should give callers the opportunity to choose whether to include that group or not. I don't feel strong about it as well.

Reply:
Makes sense. Thanks!